expression/aggregation/base_func.go

```diff
@@ -132,6 +136,45 @@ func (a *baseFuncDesc) typeInfer4ApproxCountDistinct(ctx sessionctx.Context) {
 	a.typeInfer4Count(ctx)
 }
 
+func (a *baseFuncDesc) typeInfer4ApproxPercentile(ctx sessionctx.Context) error {
+	if len(a.Args) != 2 {
+		return errors.New("APPROX_PERCENTILE should take 2 arguments")
+	}
+
+	if !a.Args[1].ConstItem(ctx.GetSessionVars().StmtCtx) {
+		return errors.New(fmt.Sprintf("APPROX_PERCENTILE should take a constant expression as percentage argument."))
```

Comment:
Unnecessary fmt.Sprintf.

Reply:
By the way, have you tested the user variable for the second argument of percentage function?

Reply:
User variable cannot be evaluated as a constant expression.
![image](https://user-images.githubusercontent.com/22445410/93849295-64110e00-fcde-11ea-97c8-6dce0044ceec.png)
Do you have any idea about this?